### PR TITLE
Add deprecation warning to golint, govet and gometalinter

### LINKED ISSUE #483: Deprecate Go Meta Linter
https://github.com/alecthomas/gometalinter

### LINKED ISSUE #485: Deprecate golint and govet
Related to #483

### PR DESCRIPTION
**Pull request by mpls104**
Fix #483
Fix #485

### CODE REVIEW

**Comment by ybiquitous** [comment]
@mpls104 Thank you for opening this PR! I've updated the PR title and description to clarify it.
Is this PR ready to review?

**Comment by mpls104** [comment]
@ybiquitous 
I haven't check the reason CI is failing but another part is ready for review!
FYI, I've use rbprettier to processor.rb and it caused a lot of change. But the only thing I did was to add one function for warning.

**Comment by ybiquitous** [comment]
@mpls104 OK, please ignore the status of SonarCloud and LGTM!

> FYI, I've use rbprettier to processor.rb and it caused a lot of change. But the only thing I did was to add one function for warning.

Umm..., the formatting looks good, but could you please do it on another PR?
Because `git blame` will be hard to track. 😓

**Comment by mpls104** [comment]
@ybiquitous 
>Umm..., the formatting looks good, but could you please do it on another PR?
Because git blame will be hard to track. 😓

Absolutely right 😓 
I revised commit so could you check again?

File: test/smokes/go_vet/expectations.rb [hunk 1]
```diff
@@ -18,5 +18,16 @@
...
-  }
+  },
+  warnings: [
+    {
+      message: <<~MSG
+        DEPRECATION WARNING!!!
+        The support for go_vet is deprecated. Sider will drop these versions on March 31, 2020.
+        Please consider to use an alternative tool GolangCi-Lint.
+MSG
+        .strip,
```
Comment (ybiquitous) [inline test/smokes/go_vet/expectations.rb]: [suggestion] What about this for code style? ⬇️ 

```ruby
message: <<~MSG.strip,
  DEPRECATION WARNING!!!
  ...
MSG
file: "sider.yml"
```

File: lib/runners/processor.rb [hunk 2]
```diff
@@ -226,6 +226,15 @@ def add_warning_if_deprecated_options(keys, doc:)
       end
     end
 
+    def add_warning_for_deprecated_linter(deadline: nil, alternative: nil)
+      deadline_str = deadline ? deadline.strftime("on %B %-d, %Y") : "in the near future"
+      add_warning <<~MSG.strip, file: ci_config_path_name
+        DEPRECATION WARNING!!!
+        The support for #{analyzer_name} is deprecated. Sider will drop these versions #{deadline_str}.
+        Please consider to use an alternative tool #{alternative}.
```
Comment (ybiquitous) [inline lib/runners/processor.rb]: [suggestion] What about fixing grammar like this?

```suggestion
        Please consider using an alternative tool #{alternative}.
```

File: lib/runners/processor/go_vet.rb [hunk 3]
```diff
@@ -26,6 +26,7 @@ def analyze(changes)
         './...',
         '--deadline=1200s'
       )
+      add_warning_for_deprecated_linter(deadline: Time.new(2_020, 3, 31), alternative: "GolangCi-Lint")
```
Comment (ybiquitous) [inline lib/runners/processor/go_vet.rb]: [suggestion] Could you please use `setup` method like this?

```ruby
def setup
  add_warning_for_deprecated_linter(deadline: Time.new(2_020, 3, 31), alternative: "GolangCi-Lint")
  yield
end
```

For also other tools.

**Comment by ybiquitous** [comment]
hadolint build sometimes is unsafe. Please retry the build when it fails. 🙏

File: lib/runners/processor.rb (hunk 2 above)
Comment (mpls104) [reply lib/runners/processor.rb]: Thanks 💦

**Pull request merged**